src/bidiMapper/domains/script/ChannelProxy.ts

```diff
@@ -56,8 +56,12 @@ export class ChannelProxy {
 
   /** Gets a ChannelProxy from window and returns its handle. */
   async startListenerFromWindow(realm: Realm, eventManager: EventManager) {
-    const channelHandle = await this.#getHandleFromWindow(realm);
-    void this.#startListener(realm, channelHandle, eventManager);
+    try {
+      const channelHandle = await this.#getHandleFromWindow(realm);
+      void this.#startListener(realm, channelHandle, eventManager);
```

Comment:
If this fails it will throw unhandled as its never awaited.

Reply:
#startListener never fails.

Reply:
Approving but it should not have a `async` as it never resolves as well.